### ISSUE TITLE
Discussion: Setting up client hints

### ISSUE THREAD

**Issue by dvtkrlbs** [Synphonyte/leptos-use#121]
[Client hints](https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints) currently experimental. Maybe we can implement some features of it using a feature flag. Especially the [Sec-CH-Prefers-Color-Scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-Prefers-Color-Scheme#browser_compatibility) header could be useful to setup dark mode on the server to prevent flash on non dark mode content.

**Comment by maccesch** [comment]
Sounds like a good idea. First we have to add a PR to `web_sys` to support the API.

**Comment by dvtkrlbs** [comment]
@maccesch from what I read it doesn't need any kind of client side feaute. Since everything only via headers. We just need the integration via axum since at first response server responds with critical hints and client retries the request.

**Comment by maccesch** [comment]
For #101 we only need the server side part indeed because the client side is already handled. But even in that case we're probably interested in being able to read it on the client as well.

For a general purpose client hints function we definitely need client side. Pages should work the same as much as possible between being rendered on the server or on the client.

This is the client API: https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API

**Comment by maccesch** [comment]
I made a PR for web_sys: https://github.com/rustwasm/wasm-bindgen/pull/3989

**Comment by dvtkrlbs** [comment]
@maccesch these are the user agent client hints api and I was mostly considerign [User preference media features client hints](https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints#user_preference_media_features_client_hints) css features where you can get for example [Sec-CH-Prefers-Reduced-Motion](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-Prefers-Reduced-Motion), [Sec-CH-Prefers-Color-Scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-Prefers-Color-Scheme). These already have client side apis.

**Comment by maccesch** [comment]
Yes, but while we're at it we might as well do it right, don't you agree?

**Comment by maccesch** [comment]
@dvtkrlbs I finally fully understood what you meant with existing client APIs. The [Sec-CH-Prefers-Color-Scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-Prefers-Color-Scheme#browser_compatibility) is now implemented.

**Comment by dvtkrlbs** [comment]
> @dvtkrlbs I finally fully understood what you meant with existing client APIs. The [Sec-CH-Prefers-Color-Scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-CH-Prefers-Color-Scheme#browser_compatibility) is now implemented.

Thank you very much

**Comment by EvanCarroll** [comment]
If this is implemented can this issue be closed?
